wremnants/CardTool.py

```diff
@@ -99,15 +145,18 @@ def allMCProcesses(self):
     def mirrorNames(self, baseName, size, offset=0):
         names = [""]*offset + [f"{baseName.format(i=i%size)}{'Up' if i % 2 else 'Down'}" for i in range(size*2)]
         return names
-
+    
     def addLnNSystematic(self, name, size, processes):
-        self.lnNSystematics.update({name : {"size" : size, "processes" : processes}})
+        if not self.isExcludedNuisance(name):
+            self.lnNSystematics.update({name : {"size" : size, "processes" : processes}})
 
     def addSystematic(self, name, systAxes, outNames=None, skipEntries=None, labelsByAxis=None, 
                         baseName="", mirror=False, scale=1, processes=None, group=None, noConstraint=False,
-                        action=None, actionArgs={}, systNameReplace=[], groupFilter=None):
+                        action=None, actionArgs={}, systNameReplace=[], groupFilter=None, passToFakes=False):
         if not processes:
-            processes = self.predictedProcesses()
+            processes = self.allMCProcesses()
+        if passToFakes and self.getFakeName() not in processes:
```

Comment:
I don't really get why this is needed. You specify the processes anyway, why not just include the fakes explicitly?

Reply:
The fact is that in some case you need only W+fakes, or W+Z+fakes, or all W/Z with fakes, and in setupComineWmass we were defining N lists of processes, and an additional one adding fakes to that. So I decided to have an additional argument, passToFakes,  in the function that create a systematic variation for the card, so you can play with it without touching the nominal list of processes. It just seemed more convenient. You can still pass the list of all predicted processes, in which case this line makes sure you don't include fakes twice

Reply:
Ok, if you prefer